File: src/main/java/com/github/t1/yaml/model/Directive.java
package com.github.t1.yaml.model;

import lombok.Value;

@Value
public class Directive {
    public static final Directive YAML_VERSION = new Directive("YAML", "1.2");

    private String name;
    private String parameters;

    boolean matchName(Directive that) { return name.equals(that.name); }
}

File: src/main/java/com/github/t1/yaml/model/Directive.kt
package com.github.t1.yaml.model

data class Directive(
    val name: String? = null,
    val parameters: String? = null
) {

    fun matchName(that: Directive): Boolean = name == that.name

    companion object {
        val YAML_VERSION = Directive("YAML", "1.2")
    }
}
